test/e2e-olm/elasticsearch_test.go

```diff
@@ -220,6 +220,16 @@ func elasticsearchFullClusterTest(t *testing.T, f *framework.Framework, ctx *fra
 		}
 	}
 
+	// Revert incorrent master node count
+	if err = f.Client.Get(goctx.TODO(), exampleName, exampleElasticsearch); err != nil {
+		return fmt.Errorf("failed to get exampleElasticsearch: %v", err)
+	}
+	exampleElasticsearch.Spec.Nodes[1].NodeCount = int32(1)
+	err = f.Client.Update(goctx.TODO(), exampleElasticsearch)
+	if err != nil {
+		return fmt.Errorf("could not update exampleElasticsearch with an additional statefulset replica: %v", err)
+	}
+
```

Comment:
@ewolinetz PTAL, AFAICS we try to test the no-4th-master-case in L195, however we never revert back the CR change. Thus, for what ever reasons most of the time the reconciliation is stuck due to an invalid master node count condition. I am not sure if this a smell in the code base, because some times this test passes. OTOH, reverting the CR to a correct state is IMHO also valid. Adding the highlighted lines let 10 of 10 runs pass. If not, I see regularly:

```
time="2020-06-29T11:40:22Z" level=info msg="Beginning full cluster restart for cert redeploy on elasticsearch"                                                                           [4/67]
time="2020-06-29T11:40:24Z" level=info msg="Timed out waiting for elasticsearch-cdm-rq90k3q5-1 to leave the cluster"                                                                           
time="2020-06-29T11:40:25Z" level=info msg="Timed out waiting for elasticsearch-cdm-rq90k3q5-2 to leave the cluster"                                                                           
{"level":"error","ts":1593430825.5387123,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"elasticsearch-controller","request":"e2e-test-19835/elasticsearch","er
ror":"Failed to reconcile Elasticsearch deployment spec: Invalid master nodes count. Please ensure there are no more than 3 total nodes with master roles","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/src/github.com/openshift/elasticsearch-operator/vendor/github.com/go-logr/zapr/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/src/github.com/openshift/elasticsearch-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:258\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/src/github.com/openshift/elasticsearch-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/src/github.com/openshift/elasticsearch-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/go/src/github.com/openshift/elasticsearch-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/go/src/github.com/openshift/elasticsearch-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/src/github.com/openshift/elasticsearch-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/src/github.com/openshift/elasticsearch
-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90"}
```

Reply:
Hmm... unfortunately that didn't help a lot on CI. Any clue?

Reply:
i agree, we need to revert that replica setting, the incorrect scale up used to be the very last test in this file so we didn't need to revert.

Reply:
@ewolinetz Do you agree that this test should be split in multiple independent ones? Is this also part of your refactoring efforts?

Reply:
I think it could help make it less fragile, yes. however it is not currently on my radar for refactoring -- I added a task here: https://issues.redhat.com/browse/LOG-817